File: Development/Assessment 1/StVincentCurrentAndForecast.js
class StVincentCurrentAndForecast {
    constructor (temperature,rainfall,tides,timerange,humidity,date,pressure,synopsis,warning,sunset,sea_conditions,winds_desc,forecast,general_conditions,sunrise) {
        this.temperature = temperature
        this.rainfall = rainfall
        this.tides = tides
        this.timerange = timerange
        this.humidity = humidity
        this.date = date
        this.pressure = pressure
        this.synopsis = synopsis
        this.warning = warning
        this.sunset = sunset
        this.sea_conditions = sea_conditions
        this.winds_desc = winds_desc
        this.forecast = forecast
        this.general_conditions = general_conditions
        this.sunrise = sunrise
    }

    toDesc () {
        str += ""
        for (var p in self) {
            if (self.hasOwnProperty(p)) {
                str += p +" is " + self.p
            }
        }
    }
}

File: Workspace/Development/Assessment 1/StVincentCurrentAndForecast.py
import json

# Create Class Definition
class StVincentCurrencyAndForecast:
    # Main Constructor
    def __init__ (self,temperature,rainfall,tides,timerange,humidity,date,pressure,synopsis,warning,sunset,sea_conditions,winds_desc,forecast,general_conditions,sunrise):
        self.temperature = temperature
        self.rainfall = rainfall
        self.tides = tides
        self.timerange = timerange
        self.humidity = humidity
        self.date = date
        self.pressure = pressure
        self.synopsis = synopsis
        self.warning = warning
        self.sunset = sunset
        self.sea_conditions = sea_conditions
        self.winds_desc = winds_desc
        self.forecast = forecast
        self.general_conditions = general_conditions
        self.sunrise = sunrise

    def toHTML (self):
        args = vars(self)
        # Table Headings
        str = "<table><thead><tr>"
        for item in args:
           str += "<th>"+ item +"</th>"
        str += "</tr></thead>"

        # Table Body
        str += "<tbody>"
        for item in args:
            str += "<tr><td>" + self.args + "</tr><td>"
        str += "</tbody></table>"